apps/dapp/frontend/components/ai/insights-dashboard.tsx

```
@@ -0,0 +1,482 @@
+
+  const allItems: Array<{ type: "recommendation" | "projection" | "coaching" | "digest"; data: PortfolioInsight | ProjectionData | CoachingContent | DigestContent }> = useMemo(() => {
+    const items: typeof allItems = []
+
+    if (filteredInsights.length > 0) {
+      for (const insight of filteredInsights) {
+        items.push({ type: "recommendation", data: insight })
+      }
+    }
+
+    return items
+  }, [filteredInsights])
```

ESLint

[error] Compilation Skipped: Existing memoization could not be preserved

React Compiler has skipped optimizing this component because the existing manual memoization could not be preserved. This value was memoized in source but not in compilation output.

/home/jailuser/git/apps/dapp/frontend/components/ai/insights-dashboard.tsx:195:169
  193 |   }, [insightsQuery.data, dismissedIds])
  194 |
> 195 |   const allItems: Array<{ type: "recommendation" | "projection" | "coaching" | "digest"; data: PortfolioInsight | ProjectionData | CoachingContent | DigestContent }> = useMemo(() => {
      |                                                                                                                                                                         ^^^^^^^^^^^^^^^
> 196 |     const items: typeof allItems = []
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 197 |
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 204 |     return items
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 205 |   }, [filteredInsights])
      | ^^^^^^^^^^^^^^^^^^^^^^^^^ Could not preserve existing memoization
  206 |
  207 |   const tabAllItems = allItems
  208 |

(react-hooks/preserve-manual-memoization)